File: python/tld_extended_attributes.py
#!/usr/bin/env python3

# This script assumes an account token.
#
# If all you have is a user token you can run this script, by passing email and password when creating the client
# like so:
# `client = Client(sandbox=True, email=<your_email>, password=<your_password>`

import sys

from auth_token import token
from dnsimple import Client

"""
Construct a client instance.

If you want to connect to production omit the sandbox option
"""
client = Client(sandbox=True, access_token=token)

"""
dnsimple.client.tlds.get_tld_extended_attributes is the method to retrieve the extended attributes
for a particular TLD.
"""
response = client.tlds.get_tld_extended_attributes(str(sys.argv[1]))

for attribute in response.data:
    print(f'- {attribute.name} ({attribute.description}): required? {attribute.required}')

File: rust/src/bin/tld_extended_attributes.rs
use dnsimple::dnsimple::new_client;
use dnsimple_rust_sample_code::token_from_file;
use std::env;

fn main() {
    let token = token_from_file();
    let args: Vec<String> = env::args().collect();

    tld_extended_attributes(&*token, &args[1]);
}

pub fn tld_extended_attributes(token: &str, tld: &str) {
    // Construct a client instance
    //
    // If you want to connect to production set the sandbox argument to false
    let client = new_client(true, token.to_string());

    // get_extended_attributes is the function to retrieve the extended attributes for a particular
    // TLD.
    let response = client.tlds().get_tld_extended_attributes(tld.to_string());

    println!(
        "The extended attributes for tld {}: {:?}",
        tld,
        response.unwrap().data.unwrap()
    );
}
